examples/showcase/anatomy.py

```diff
@@ -0,0 +1,143 @@
+# This figure shows the name of several matplotlib elements composing a figure
+
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.patches import Circle
+from matplotlib.ticker import MultipleLocator, FuncFormatter
+
+np.random.seed(123)
+
+X = np.linspace(0.5, 3.5, 100)
+Y1 = 3+np.cos(X)
+Y2 = 1+np.cos(1+X/0.75)/2
+Y3 = np.random.uniform(Y1, Y2, len(X))
+
+plt.figure(figsize=(8, 8), facecolor="w")
+ax = plt.subplot(1, 1, 1, aspect=1)
+
+
+def minor_tick(x, pos):
+    if not x % 1.0:
+        return ""
+    return "%.2f" % x
+
+plt.axes().xaxis.set_major_locator(MultipleLocator(1.000))
+plt.axes().xaxis.set_minor_locator(MultipleLocator(0.250))
+plt.axes().yaxis.set_major_locator(MultipleLocator(1.000))
+plt.axes().yaxis.set_minor_locator(MultipleLocator(0.250))
+plt.axes().xaxis.set_minor_formatter(FuncFormatter(minor_tick))
+
+plt.xlim(0, 4)
+plt.ylim(0, 4)
+
+plt.tick_params(which='major', width=1.0)
+plt.tick_params(which='major', length=10)
+plt.tick_params(which='minor', width=1.0, labelsize=10)
+plt.tick_params(which='minor', length=5, labelsize=10, labelcolor='0.25')
+
+plt.grid(linestyle="--", linewidth=0.5, color='.25', zorder=-10)
+
+plt.plot(X, Y1, c=(0.25, 0.25, 1.00), lw=2, label="Blue signal", zorder=10)
+plt.plot(X, Y2, c=(1.00, 0.25, 0.25), lw=2, label="Red signal")
+plt.scatter(X, Y3, c='w')
+
+plt.title("Anatomy of a figure", fontsize=20)
+plt.xlabel("X axis label")
+plt.ylabel("Y axis label")
+
+plt.legend(frameon=False)
+
+
+def circle(x, y, radius=0.15):
+    center = x, y
+    circle = Circle(center, radius, clip_on=False, zorder=10,
+                    edgecolor='white', facecolor='none', linewidth=5.0)
+    plt.axes().add_artist(circle)
+    circle = Circle(center, radius, clip_on=False, zorder=20,
+                    edgecolor='none', facecolor='black', alpha=.025)
+    plt.axes().add_artist(circle)
+    circle = Circle(center, radius, clip_on=False, zorder=30,
+                    edgecolor='black', facecolor='none', linewidth=1.0)
```

Comment:
I think all of this could be simplified to one `Circle` call with help of `PathEffects.withStroke` http://matplotlib.org/examples/pylab_examples/patheffect_demo.html